pkg/vz/vm_darwin.go

```diff
@@ -0,0 +1,493 @@
+package vz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/Code-Hex/vz/v3"
+	"github.com/docker/go-units"
+	"github.com/lima-vm/lima/pkg/driver"
+	"github.com/lima-vm/lima/pkg/iso9660util"
+	"github.com/lima-vm/lima/pkg/limayaml"
+	"github.com/lima-vm/lima/pkg/localpathutil"
+	"github.com/lima-vm/lima/pkg/networks"
+	"github.com/lima-vm/lima/pkg/store/filenames"
+	"github.com/sirupsen/logrus"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+func startVM(ctx context.Context, driver *driver.BaseDriver) (*vz.VirtualMachine, error) {
+	server, client, err := createSockPair()
+	if err != nil {
+		return nil, err
+	}
+	machine, err := createVM(driver, client)
+	if err != nil {
+		return nil, err
+	}
+
+	fileConn, err := net.FileConn(server)
+	if err != nil {
+		return nil, err
+	}
+
+	err = machine.Start()
+
+	networks.StartGVisor(ctx, &networks.GVisorOpts{
+		Conn:         fileConn,
+		MTU:          1500,
+		SSHLocalPort: driver.SSHLocalPort,
+		MacAddress:   limayaml.MACAddress(driver.Instance.Dir),
+		Stream:       false,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		//Handle errors via errCh and handle stop vm during context close
+		_, cancelHa := context.WithCancel(ctx)
+
+		for {
+			select {
+			case <-ctx.Done():
+				logrus.Println("Context closed, stopping vm")
+				if machine.CanStop() {
+					_, err := machine.RequestStop()
+					logrus.Errorf("Error while stopping the VM %q", err)
+				}
+			case newState := <-machine.StateChangedNotify():
+				if newState == vz.VirtualMachineStateRunning {
```

Comment:
Can we use `switch{}` for `newState` comparisons?

Reply:
Done